src/meta/src/controller/utils.rs

```diff
@@ -1177,6 +1180,170 @@ pub fn extract_external_table_name_from_definition(table_definition: &str) -> Op
+        ObjectType::Sink => rename_relation!(Sink, sink, sink_id, object_id),
+        ObjectType::Subscription => {
+            rename_relation!(Subscription, subscription, subscription_id, object_id)
+        }
+        ObjectType::View => rename_relation!(View, view, view_id, object_id),
+        ObjectType::Index => {
+            let (mut index, obj) = Index::find_by_id(object_id)
+                .find_also_related(Object)
+                .one(txn)
+                .await?
+                .unwrap();
+            index.name = object_name.into();
+            let index_table_id = index.index_table_id;
+            let old_name = rename_relation!(Table, table, table_id, index_table_id);
+
+            // the name of index and its associated table is the same.
+            let active_model = index::ActiveModel {
+                index_id: sea_orm::ActiveValue::Set(index.index_id),
+                name: sea_orm::ActiveValue::Set(object_name.into()),
+                ..Default::default()
+            };
+            active_model.update(txn).await?;
+            to_update_relations.push(PbRelation {
+                relation_info: Some(PbRelationInfo::Index(
+                    ObjectModel(index, obj.unwrap()).into(),
+                )),
+            });
+            old_name
+        }
+        _ => unreachable!("only relation name can be altered."),
+    };
+
+    Ok((to_update_relations, old_name))
+}
+
+/// `rename_relation_refer` updates the definition of relations that refer to the target one,
+/// it commits the changes to the transaction and returns all the updated relations.
+pub async fn rename_relation_refer(
+    txn: &DatabaseTransaction,
+    object_type: ObjectType,
+    object_id: ObjectId,
+    object_name: &str,
+    old_name: &str,
+) -> MetaResult<Vec<PbRelation>> {
+    use sea_orm::ActiveModelTrait;
+
+    use crate::controller::rename::alter_relation_rename_refs;
+
+    let mut to_update_relations = vec![];
+    macro_rules! rename_relation_ref {
+        ($entity:ident, $table:ident, $identity:ident, $object_id:expr) => {{
+            let (mut relation, obj) = $entity::find_by_id($object_id)
+                .find_also_related(Object)
+                .one(txn)
+                .await?
+                .unwrap();
+            relation.definition =
+                alter_relation_rename_refs(&relation.definition, old_name, object_name);
+            let active_model = $table::ActiveModel {
+                $identity: Set(relation.$identity),
+                definition: Set(relation.definition.clone()),
+                ..Default::default()
+            };
+            active_model.update(txn).await?;
+            to_update_relations.push(PbRelation {
+                relation_info: Some(PbRelationInfo::$entity(
+                    ObjectModel(relation, obj.unwrap()).into(),
+                )),
+            });
+        }};
+    }
+    let mut objs = get_referring_objects(object_id, txn).await?;
```

Comment:
So the system table `object_dependency` didn't store the tables that being used by table sinks. Is it expected? 

This looks error-prone to me because every time we look for referring objects, such as dropping or renaming, these `sink into table` should be considered as well, as here we did it in L1299-1319.

Reply:
Yes, it is a historical issue. I will open a PR to manage them uniformly with `object_dependency`.